src/main.ts

```diff
@@ -66,6 +66,14 @@ if (process.env.NODE_ENV !== 'development') {
   Sentry.init({
     dsn: SENTRY_DSN,
     integrations: [Sentry.electronMinidumpIntegration()],
+
+    // conditionally send the event based on the user's settings
+    beforeSend: (event) => {
+      if (appSettings.telemetry.errorReport) {
+        return event
+      }
```

Comment:
Sentry is always initialized in Prod mode, but the event is dynamically sent based on the `errorReport` setting.